admin/services/data-access/group.data.service.js

```diff
@@ -20,17 +24,99 @@ groupDataService.getGroups = async function (query) {
     .exec()
 }
 
+/**
+ * Get active groups (non-soft-deleted).
+ * @returns {Promise<*>}
+ */
+groupDataService.sqlFindGroups = async (schoolId) => {
+  const sql = `
+  SELECT g.id, g.name, COUNT(pg.pupil_id) as pupilCount 
+  FROM ${sqlService.adminSchema}.[group] g
+  LEFT OUTER JOIN ${sqlService.adminSchema}.pupilGroup pg 
+  ON g.id = pg.group_id
+  WHERE g.isDeleted=0
+  AND g.school_id=@schoolId
+  GROUP BY g.id, g.name
+  ORDER BY name ASC`
+  const params = [
+    {
+      name: 'schoolId',
+      value: schoolId,
+      type: TYPES.Int
+    }
+  ]
+  return sqlService.query(sql, params)
+}
+
 /**
  * Get group document by _id.
+ * @deprecated use sqlFindGroup
  * @param query
  * @returns {Promise<*>}
  */
 groupDataService.getGroup = async function (query) {
   return Group.findOne(query).lean().exec()
 }
 
+/**
+ * Get group by id.
+ * @param groupId
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlFindGroupById = async (groupId, schoolId) => {
```

Comment:
I would prefer it to be name `sqlFindOneById`  - the one indicates it returns an object and not an array.  The `Group` bit is redundant as this is the GroupDataService.